avogadro/qtplugins/bonding/bonding.cpp

```diff
@@ -121,7 +130,30 @@ void Bonding::bond2()
 
 void Bonding::clearBonds()
 {
-  m_molecule->clearBonds();
+  // remove any bonds connected to the selected atoms
```

Comment:
This felt asymmetric when I tested. Removing bonds remove from all if nothing was selected but would not add back to all if nothing was selected. Also, removing from selected removed from the any bond that had a selected atom at either end, whereas adding back only added bonds where both ends were selected.